library/src/main/java/com/novoda/downloadmanager/lib/OkHttpNotificationImageRetriever.java

```diff
@@ -28,7 +38,10 @@ public Bitmap retrieveImage(String imageUrl) {
             Response response = client.newCall(request).execute();
             InputStream inputStream = response.body().byteStream();
             try {
-                return BitmapFactory.decodeStream(inputStream);
+                this.imageUrl = imageUrl;
```

Comment:
Moved it to after the bitmap is decoded. The earlier we do it, the more chance of caching an invalid bitmap.